src/services/pocket-relayer.ts

```diff
@@ -848,14 +821,40 @@ export class PocketRelayer {
     )
   }
 
-  static async stickyRelayResult(
-    preferredNodeAddress: string | undefined,
-    relayNodePublicKey: string
-  ): Promise<string> {
-    if (!preferredNodeAddress) {
-      return 'NONE'
+  async setStickinessKey(
+    { duration, keyPrefix, rpcID }: StickinessOptions,
+    blockchainID: string,
+    applicationID: string,
+    nodeAddress: string,
+    data: string
+  ): Promise<void> {
+    if (!keyPrefix && !rpcID) {
+      return
     }
 
-    return preferredNodeAddress === (await getAddressFromPublicKey(relayNodePublicKey)) ? 'SUCCESS' : 'FAILURE'
+    // If no key prefix is given, set based on rpcID
+    if (keyPrefix) {
+      const clientStickyKey = `${keyPrefix}-${this.ipAddress}-${blockchainID}`
+
+      const nextRequest = await this.redis.get(clientStickyKey)
```

Comment:
Another comment here please -- explain that if the key is still set, we don't want to set it again and extend the stickiness